sklearn/linear_model/_coordinate_descent.py

```diff
@@ -1649,8 +1658,20 @@ def fit(self, X, y, sample_weight=None):
                 for l1_ratio in l1_ratios
             ]
         else:
+            # Making sure alphas entries are scalars.
+            if np.isscalar(alphas):
+                check_scalar_alpha(alphas, "alphas")
+            else:
+                # alphas is an iterable item in this case.
+                n_alphas = 1 if np.ndim(alphas) == 0 else len(alphas)
+                if n_alphas != 1:
+                    for index, alpha in enumerate(alphas):
+                        check_scalar_alpha(alpha, f"alphas[{index}]")
+                else:
+                    check_scalar_alpha(alphas[0], "alphas")
```

Comment:
With the `np.isscalar(alphas)` check, I think `np.ndim(alphas) != 0` all the time, which means this can be check can be simplified:

```suggestion
                # alphas is an iterable item in this case.
                for index, alpha in enumerate(alphas):
                    check_scalar_alpha(alpha, f"alphas[{index}]")
```